core/templates/hash_map.h

```diff
@@ -106,9 +106,25 @@ class HashMap {
 
 	bool _lookup_pos_with_hash(const TKey &p_key, uint32_t &r_pos, uint32_t p_hash) const {
 		uint32_t pos = p_hash & capacity;
-		uint32_t distance = 0;
 
+		if (hashes[pos] == p_hash && Comparator::compare(elements[pos]->data.key, p_key)) {
+			r_pos = pos;
+			return true;
+		}
+
+		if (hashes[pos] == EMPTY_HASH) {
+			return false;
+		}
+
```

Comment:
I added this because the first iteration always checked the condition `distance > _get_probe_length(pos, hashes[pos])` that was `false`.